main.go

```diff
@@ -297,54 +303,107 @@ func enableDevworkspaceSupport(mgr manager.Manager) error {
 		return err
 	}
 
-	// we only enable Devworkspace support, if there is the controller.devfile.io resource group in the cluster
-	// we assume that if the group is there, then we have all the expected CRs there, too.
-
-	cl, err := discovery.NewDiscoveryClientForConfig(mgr.GetConfig())
+	// we only enable DevWorkspace support, if there is the controller.devfile.io resource group in the cluster
+	// and DevWorkspaces are enabled at least on one CheCluster
+	dwEnabled, err := doesCheClusterWithDevWorkspaceEnabledExist(mgr)
 	if err != nil {
 		return err
 	}
 
-	groups, err := cl.ServerGroups()
+	if !dwEnabled {
+		doAsyncRestartWhenDevWorkspaceEnabled(mgr, stop)
+		return nil
+	}
+
+	// we assume that if the group is there, then we have all the expected CRs there, too.
+	dwApiExists, err := findApiGroup(mgr, "controller.devfile.io")
 	if err != nil {
 		return err
 	}
 
-	supported := false
-	for _, g := range groups.Groups {
-		if g.Name == "controller.devfile.io" {
-			supported = true
-			break
-		}
+	if !dwApiExists {
+		return errors.New("there is a CheCluster with DevWorkspace enabled but " +
+			"devworkspace api group 'controller.devfile.io' is not available")
 	}
 
-	if supported {
-		if err := dwo_api.AddToScheme(mgr.GetScheme()); err != nil {
-			return err
-		}
+	if err := dwo_api.AddToScheme(mgr.GetScheme()); err != nil {
+		return err
+	}
 
-		routing := dwr.DevWorkspaceRoutingReconciler{
-			Client:       mgr.GetClient(),
-			Log:          ctrl.Log.WithName("controllers").WithName("DevWorkspaceRouting"),
-			Scheme:       mgr.GetScheme(),
-			SolverGetter: solver.Getter(mgr.GetScheme()),
-		}
-		if err := routing.SetupWithManager(mgr); err != nil {
-			setupLog.Error(err, "unable to set up controller", "controller", "DevWorkspaceRouting")
-			return err
-		}
+	routing := dwr.DevWorkspaceRoutingReconciler{
+		Client:       mgr.GetClient(),
+		Log:          ctrl.Log.WithName("controllers").WithName("DevWorkspaceRouting"),
+		Scheme:       mgr.GetScheme(),
+		SolverGetter: solver.Getter(mgr.GetScheme()),
+	}
+	if err := routing.SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to set up controller", "controller", "DevWorkspaceRouting")
+		return err
+	}
 
-		userNamespaceReconciler := usernamespace.NewReconciler()
+	userNamespaceReconciler := usernamespace.NewReconciler()
 
-		if err = userNamespaceReconciler.SetupWithManager(mgr); err != nil {
-			setupLog.Error(err, "unable to set up controller", "controller", "CheUserReconciler")
-			return err
+	if err = userNamespaceReconciler.SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to set up controller", "controller", "CheUserReconciler")
+		return err
+	}
+
+	setupLog.Info("DevWorkspace support enabled")
+
+	return nil
+}
+
+func doesCheClusterWithDevWorkspaceEnabledExist(mgr manager.Manager) (bool, error) {
+	cheClusters := &orgv1.CheClusterList{}
+	err := mgr.GetClient().List(context.TODO(), cheClusters, &client.ListOptions{})
```

Comment:
it does not work at this stage.
```
2021-10-07T18:12:24.344+0300	ERROR	setup	unable to initialize devworkspace support	{"error": "the cache is not started, can not read objects"}
main.main
```
@tolusha @metlos any ideas how to do that check at this point?

Alternative - we can drop backward compatibility fallback https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/dev-workspace/dev_workspace.go#L88
and just check DevWorkspace allowed env var, as @metlos proposed.

Reply:
Starting manager is a blocking operation, so at start up point we don't have better alternative than using non-cached client.